Tella/Scenes/Settings/ViewModel/GDriveAuthViewModel.swift

```diff
@@ -12,21 +12,23 @@ import Combine
 class GDriveAuthViewModel: ObservableObject {
     private let gDriveRepository: GDriveRepositoryProtocol
     private var cancellables = Set<AnyCancellable>()
-
-    init(repository: GDriveRepositoryProtocol = GDriveRepository.shared) {
+    @Published var signInState: ViewModelState<String?> = .loaded(nil)
+    init(repository: GDriveRepositoryProtocol) {
         self.gDriveRepository = repository
     }
 
     func handleSignIn(completion: @escaping () -> Void) {
+        self.signInState = .loading
         Task {
             do {
                 try await gDriveRepository.handleSignIn()
                 DispatchQueue.main.async {
+                    self.signInState = .loaded(nil)
                     completion()
                 }
             } catch let error {
                 DispatchQueue.main.async {
-                    Toast.displayToast(message: error.localizedDescription)
+                    self.signInState = .error(error.localizedDescription)
                 }
             }
         }
```

Comment:
You can replace `DispatchQueue.main.async {` with `@MainActor` 
and the method will look like this: 
   ```
 func handleSignIn(completion: @escaping () -> Void) {
        self.signInState = .loading
        Task { @MainActor in
            do {
                try await gDriveRepository.handleSignIn()
                    self.signInState = .loaded(nil)
                    completion()
            } catch let error {
                    self.signInState = .error(error.localizedDescription)
            }
        }
    }

```